File: courses/codes/TestLight.java
public class TestLight {

    public static void main(String args[]) {

        System.out.println("Running Light tests...");
        // 1. Test Light()
        Light n = new Light();
        assert n.toString().equals("on white\tnot burnt out");

        // 2. Test Light(o, b, c)
        n = new Light(false, true, "green");
        assert n.toString().equals("off green\tburnt out");

        // 3. Test burn_out()
        n = new Light();
        n.burnOut();
        assert n.toString().equals("off white\tburnt out");

        // 4. Test flip()
        n = new Light(true, false, "white");
        assert n.toString().equals("on white\tnot burnt out");
        n.flip();
        assert n.toString().equals("off white\tnot burnt out");
        n.flip();
        assert n.toString().equals("on white\tnot burnt out");
        n.burnOut();
        n.flip();
        assert n.toString().equals("off white\tburnt out");

        // 5. Test is_on()
        n = new Light();
        assert n.isOn();
        n.flip();
        assert !n.isOn();

        // 6. Test get_color()
        n = new Light(true, false, "red");
        assert n.getColor().equals("red");
        n = new Light(true, false, "green");
        assert n.getColor().equals("green");
        n = new Light(true, false, "blue");
        assert n.getColor().equals("blue");
        n = new Light(true, false, "white");
        assert n.getColor().equals("white");
        n = new Light(true, false, "purple");
        assert n.getColor().equals("white");

        // 7. Test set_color()
        n = new Light();
        n.setColor("red");
        assert n.getColor().equals("red");
        n.setColor("green");
        assert n.getColor().equals("green");
        n.setColor("blue");
        assert n.getColor().equals("blue");
        n.setColor("white");
        assert n.getColor().equals("white");
        n.setColor("purple");
        assert n.getColor().equals("white");

        System.out.println("Tests passed.");
    }
}

File: courses/codes/python-class/test_light.py
from Light import Light

print("Running Light tests...")

# 1. Test Light()
n = Light()
assert str(n) == "on white\tnot burnt out"

# 2. Test Light(o, b, c)
n = Light(o=False, b=True, c="green")
assert str(n) == "off green\tburnt out"

# 3. Test burn_out()
n = Light()
n.burn_out()
assert str(n) == "off white\tburnt out"

# 4. Test flip()
n = Light(o=True, b=False, c="white")
assert str(n) == "on white\tnot burnt out"
n.flip()
assert str(n) == "off white\tnot burnt out"
n.flip()
assert str(n) == "on white\tnot burnt out"
n.burn_out()
n.flip()
assert str(n) == "off white\tburnt out"

# 5. Test is_on()
n = Light()
assert n.is_on()
n.flip()
assert not n.is_on()

# 6. Test get_color()
n = Light(c="red")
assert n.get_color() == "red"
n = Light(c="green")
assert n.get_color() == "green"
n = Light(c="blue")
assert n.get_color() == "blue"
n = Light(c="white")
assert n.get_color() == "white"
n = Light(c="purple")
assert n.get_color() == "white"

# 7. Test set_color()
n = Light()
n.set_color("red")
assert n.get_color() == "red"
n.set_color("green")
assert n.get_color() == "green"
n.set_color("blue")
assert n.get_color() == "blue"
n.set_color("white")
assert n.get_color() == "white"
n.set_color("purple")
assert n.get_color() == "white"

print("Tests passed.")




